halo1/model.py

```diff
@@ -108,13 +212,26 @@ def import_halo1_markers_from_jms(jms, *, scale=1.0, node_size=0.01,
 		if import_radius:
 			set_uniform_scale(scene_marker, marker.radius)
 
-		# Assign parent if index is valid.
-		scene_marker.parent = scene_nodes.get(marker.parent, None)
-
 		set_rotation_from_jms(scene_marker, marker)
 
 		set_translation_from_jms(scene_marker, marker, scale)
 
+		# Assign parent if index is valid.
+		parent = scene_nodes.get(marker.parent, None)
+		if armature and parent:
+			scene_marker.parent = armature
+			scene_marker.parent_type = 'BONE'
+			scene_marker.parent_bone = parent.name
+
+			# Ajust location for difference in forward axis between Halo
+			# and Blender bones.
+			scene_marker.location.rotate(Euler((0.0, 0.0, math.radians(90.0))))
+			# Adjust location for being parented to the tail end of the bone
+			# rather than the head end.
+			scene_marker.location.y -= parent.length
```

Comment:
I don't follow how these actions accomplish the goals in the comments.

Reply:
The rotation is offset by 0, 0, -90 euler, so we rotate it to fix that.

Then we subtract the length of the parent bone from the y axis, the y axis being the axis that is offset.

Halo bone behavior expects markers their 0 0 0 to be at the bone base. But when attaching in Blender its 0 0 0 is instead at the bone's tail end.

Reply:
You should put this in as a comment in the code. This made it much clearer. Only thing I'd add is *which* rotation is offset by 90?